addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -707,6 +700,67 @@ const actions = {
             dispatch('_insertMessage', preview.last_message);
         }
     },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.env
+     * @param {Object} param0.state
+     * @param {string} activityLocalId
+     * @param {Object} param2
+     * @param {string|boolean} param2.feedback
+     * @param {Array} [param2.attachmentLocalIds=[]]
+     */
+    async markActivityAsDone(
+        { dispatch, env, state },
+        activityLocalId,
+        { feedback=false, attachmentLocalIds=[] }
+    ) {
+        const activity = state.activities[activityLocalId];
+        const chatter = state.chatters[activity.chatterLocalId];
+        const attachmentIds = [];
+        for (const attachmentLocalId in state.attachments) {
+            if (attachmentLocalIds.includes(attachmentLocalId)) {
+                attachmentIds.push(state.attachments[attachmentLocalId].id);
+            }
+        }
```

Comment:
why loop on all attachments when you could loop on attachments passed as arg? Imagine a store with lots of attachments

Reply:
oopsie :astonished: